include/mapbox/util/type_wrapper.hpp

```diff
@@ -35,6 +63,7 @@ class TypeWrapper {
 
     using storage_t = std::unique_ptr<void, void (*)(void*)>;
     storage_t storage_;
+    mutable std::mutex mutex_;
```

Comment:
why does it have to be mutable?

Reply:
So the mutex can be locked inside a `const` method.